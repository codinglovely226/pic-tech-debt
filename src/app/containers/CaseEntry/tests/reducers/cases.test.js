/* eslint-disable */

import casesReducer from 'containers/CaseEntry/reducers/cases';

import {
  addCallRecord,
  addCase,
  updateApplicationKeyId,
  updateCaseOutcome,
  updateCaseStatus,
  updateCaseTopic,
  updateCaseType,
} from '../../actions/case';

import {
  saveNote,
} from 'containers/NoteEntry/actions/case';

describe('casesReducer', () => {
  var mockedDate = new Date(2018, 3, 7, 17, 8);
  global.Date = jest.fn(() => mockedDate);
  it('returns the initial state', () => {
    expect(casesReducer(undefined, {})).toMatchSnapshot();
  });
  it('add call record', () => {
    const action = addCallRecord('Test Case', {'name': 'Test SPI'});
    action.commsNoteId = 'de6d38464f8ee145f0731aca7faa24f78d07e25f';
    expect(casesReducer({}, action)).toMatchSnapshot();
  });
  it('add case', () => {
    const action = addCase({'name': 'Test SPI', 'siteId': 2, 'roles': ['SPI']}, 'WAPIC', {id: 'appKeyId', key: 'appKey'});
    action.id = '5ead68d0e8d5d48b3262b58478763e2f20c61f15';
    action.commsNoteId = 'de6d38464f8ee145f0731aca7faa24f78d07e25f';
    expect(casesReducer({}, action)).toMatchSnapshot();
  });
  it('save note', () => {
    expect(casesReducer({'Test Case': {'_attachments': {}}}, saveNote('Test Case', 'Test Note', 'Test Context', 'Test Subcontext', 'Test Title', 'Test Content', [{}], {'name': 'Test SPI'}))).toMatchSnapshot();
  });
  it('update application key id', () => {
    expect(casesReducer({'Test Case': {}}, updateApplicationKeyId('Test Case', { id: 'appKeyId', key: 'appKey' }))).toMatchSnapshot();
  });
  it('update case outcome', () => {
    expect(casesReducer({'Test Case': {}}, updateCaseOutcome('Test Case', 'Old Outcome', 'New Outcome'))).toMatchSnapshot();
  });
  it('update case status', () => {
    expect(casesReducer({'Test Case': {}}, updateCaseStatus('Test Case', 'open', {'name': 'Test SPI'}, false))).toMatchSnapshot();
  });
  it('update case topic', () => {
    expect(casesReducer({'Test Case': {}}, updateCaseTopic('Test Case', 'Old Topic', 'New Topic'))).toMatchSnapshot();
  });
  it('update case type', () => {
    expect(casesReducer({'Test Case': {'type': 'case'}}, updateCaseType('Test Case', 'hoax'))).toMatchSnapshot();
  });
});
