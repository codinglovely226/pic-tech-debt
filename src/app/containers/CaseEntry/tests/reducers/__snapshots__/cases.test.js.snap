exports[`casesReducer add call record 1`] = `
Object {
  "Test Case": Object {
    "_attachments": Object {},
    "_id": "",
    "_rev": "",
    "agents": Array [],
    "appKeyId": "",
    "careTeam": Array [],
    "communications": Array [
      Object {
        "deleted": false,
        "direction": "incoming",
        "endTimestamp": null,
        "history": Object {
          "agents": Object {},
          "features": Object {
            "descriptors": "",
            "outcome": "",
            "pastDescriptors": Object {},
            "when": "",
          },
          "investigations": Object {},
          "patient": Object {
            "age": null,
            "ageGroup": null,
            "ageUnits": null,
            "ageYears": null,
            "animalType": null,
            "category": null,
            "circumstance": null,
            "gender": null,
            "localityAddress1": null,
            "localityCity": null,
            "localityCountry": null,
            "localityGeometry": null,
            "localityISOCountry": null,
            "localityLocation": null,
            "localityName": null,
            "localityPlaceId": null,
            "localityPostcode": null,
            "localityState": null,
            "localityTelephone": null,
            "medicalRecordNumber": null,
            "medicalRecordNumberEnc": null,
            "name": null,
            "nameEnc": null,
            "occupation": null,
            "outcome": null,
            "poisonSeverityScore": null,
            "pregnantLactating": null,
            "pregnantStatus": null,
            "pregnantTrimester": null,
            "pregnantTrimesterUnknown": null,
            "riskAssessment": null,
            "topic": null,
            "weight": null,
            "weightUnits": null,
          },
          "treatments": Object {
            "pastTreatments": Object {},
            "referrals": Array [],
            "treatments": Array [],
          },
        },
        "interlocutor": Object {
          "category": "",
          "contact": "",
          "name": "",
          "nameEnc": "",
          "organisation": Object {
            "address1": "",
            "address2": "",
            "city": "",
            "country": "",
            "geometry": Object {},
            "isoCountry": "",
            "location": "",
            "name": "",
            "placeId": "",
            "postcode": "",
            "state": "",
            "telephone": "",
          },
        },
        "medium": Object {
          "detail": "",
          "detailEnc": "",
          "type": "Phone",
        },
        "noteRef": "de6d38464f8ee145f0731aca7faa24f78d07e25f",
        "sequenceId": 1,
        "spi": Object {
          "id": undefined,
          "job": undefined,
          "name": "Test SPI",
          "org": undefined,
          "roles": undefined,
          "siteId": undefined,
        },
        "timestamp": "2018-04-07T17:08:00.000Z",
        "type": "comms",
      },
    ],
    "features": Array [],
    "incidentId": "",
    "investigations": Array [],
    "naturalId": "",
    "notes": Object {
      "de6d38464f8ee145f0731aca7faa24f78d07e25f": Object {
        "attachments": Array [],
        "content": "",
        "context": "general",
        "id": "de6d38464f8ee145f0731aca7faa24f78d07e25f",
        "spi": Object {
          "id": undefined,
          "job": undefined,
          "name": "Test SPI",
          "org": undefined,
          "roles": undefined,
          "siteId": undefined,
        },
        "subcontext": "",
        "timestamp": "2018-04-07T17:08:00.000Z",
        "title": "",
      },
    },
    "outcome": "",
    "patient": Object {
      "age": "",
      "ageGroup": "adult",
      "ageUnits": "years",
      "ageYears": "",
      "animalType": "",
      "category": "human",
      "circumstance": "",
      "gender": "",
      "locality": Object {
        "address1": "",
        "address2": "",
        "city": "",
        "country": "",
        "geometry": Object {},
        "isoCountry": "",
        "location": "",
        "name": "",
        "placeId": "",
        "postcode": "",
        "state": "",
        "telephone": "",
      },
      "medicalRecordNumber": null,
      "medicalRecordNumberEnc": "",
      "name": "",
      "nameEnc": "",
      "occupation": "unknown",
      "poisonSeverityScore": "Moderate",
      "pregnant": Object {
        "lactating": "unknown",
        "status": "unknown",
        "trimester": "unknown",
        "trimesterUnknown": null,
      },
      "riskAssessment": "Moderate Risk",
      "weight": "",
      "weightUnits": "kg",
    },
    "referrals": Array [],
    "siteCreated": 1,
    "status": "open",
    "timestamp": "2018-04-07T17:08:00.000Z",
    "topic": "",
    "treatments": Array [],
    "type": "case",
  },
}
`;

exports[`casesReducer add case 1`] = `
Object {
  "5ead68d0e8d5d48b3262b58478763e2f20c61f15": Object {
    "_attachments": Object {},
    "_id": "5ead68d0e8d5d48b3262b58478763e2f20c61f15",
    "_rev": "",
    "agents": Array [],
    "appKeyId": "appKeyId",
    "careTeam": Array [
      2,
    ],
    "communications": Array [
      Object {
        "deleted": false,
        "direction": "incoming",
        "endTimestamp": null,
        "history": Object {
          "agents": Object {},
          "features": Object {
            "descriptors": "",
            "outcome": "",
            "pastDescriptors": Object {},
            "when": "",
          },
          "investigations": Object {},
          "patient": Object {
            "age": null,
            "ageGroup": Array [
              null,
              "adult",
            ],
            "ageUnits": null,
            "ageYears": null,
            "animalType": null,
            "category": Array [
              null,
              "human",
            ],
            "circumstance": null,
            "gender": null,
            "localityAddress1": null,
            "localityCity": null,
            "localityCountry": null,
            "localityGeometry": Array [
              null,
              Object {},
            ],
            "localityISOCountry": null,
            "localityLocation": null,
            "localityName": null,
            "localityPlaceId": null,
            "localityPostcode": null,
            "localityState": null,
            "localityTelephone": null,
            "medicalRecordNumber": null,
            "medicalRecordNumberEnc": null,
            "name": null,
            "nameEnc": null,
            "occupation": Array [
              null,
              "unknown",
            ],
            "outcome": null,
            "poisonSeverityScore": Array [
              null,
              "Moderate",
            ],
            "pregnantLactating": Array [
              null,
              "unknown",
            ],
            "pregnantStatus": Array [
              null,
              "unknown",
            ],
            "pregnantTrimester": Array [
              null,
              "unknown",
            ],
            "pregnantTrimesterUnknown": null,
            "riskAssessment": Array [
              null,
              "Moderate Risk",
            ],
            "topic": null,
            "weight": null,
            "weightUnits": Array [
              null,
              "kg",
            ],
          },
          "treatments": Object {
            "pastTreatments": Object {},
            "referrals": Array [],
            "treatments": Array [],
          },
        },
        "interlocutor": Object {
          "category": "",
          "contact": "",
          "name": "",
          "nameEnc": "",
          "organisation": Object {
            "address1": "",
            "address2": "",
            "city": "",
            "country": "",
            "geometry": Object {},
            "isoCountry": "",
            "location": "",
            "name": "",
            "placeId": "",
            "postcode": "",
            "state": "",
            "telephone": "",
          },
        },
        "medium": Object {
          "detail": "",
          "detailEnc": "",
          "type": "Phone",
        },
        "noteRef": "de6d38464f8ee145f0731aca7faa24f78d07e25f",
        "sequenceId": 1,
        "spi": Object {
          "id": undefined,
          "job": undefined,
          "name": "Test SPI",
          "org": undefined,
          "roles": Array [
            "SPI",
          ],
          "siteId": 2,
        },
        "timestamp": "2018-04-07T17:08:00.000Z",
        "type": "comms",
      },
    ],
    "features": Array [
      Object {
        "descriptors": Object {},
        "spi": Object {
          "id": undefined,
          "job": undefined,
          "name": "Test SPI",
          "org": undefined,
          "roles": Array [
            "SPI",
          ],
          "siteId": 2,
        },
        "timestamp": "2018-04-07T17:08:00.000Z",
        "when": "at",
      },
    ],
    "incidentId": "",
    "investigations": Array [],
    "naturalId": "WAPIC:5ead68d0",
    "notes": Object {
      "de6d38464f8ee145f0731aca7faa24f78d07e25f": Object {
        "attachments": Array [],
        "content": "",
        "context": "general",
        "id": "de6d38464f8ee145f0731aca7faa24f78d07e25f",
        "spi": Object {
          "id": undefined,
          "job": undefined,
          "name": "Test SPI",
          "org": undefined,
          "roles": Array [
            "SPI",
          ],
          "siteId": 2,
        },
        "subcontext": "",
        "timestamp": "2018-04-07T17:08:00.000Z",
        "title": "",
      },
    },
    "outcome": "",
    "patient": Object {
      "age": "",
      "ageGroup": "adult",
      "ageUnits": "years",
      "ageYears": "",
      "animalType": "",
      "category": "human",
      "circumstance": "",
      "gender": "",
      "locality": Object {
        "address1": "",
        "address2": "",
        "city": "",
        "country": "",
        "geometry": Object {},
        "isoCountry": "",
        "location": "",
        "name": "",
        "placeId": "",
        "postcode": "",
        "state": "",
        "telephone": "",
      },
      "medicalRecordNumber": null,
      "medicalRecordNumberEnc": "",
      "name": "",
      "nameEnc": "",
      "occupation": "unknown",
      "poisonSeverityScore": "Moderate",
      "pregnant": Object {
        "lactating": "unknown",
        "status": "unknown",
        "trimester": "unknown",
        "trimesterUnknown": null,
      },
      "riskAssessment": "Moderate Risk",
      "weight": "",
      "weightUnits": "kg",
    },
    "referrals": Array [],
    "siteCreated": 2,
    "status": "open",
    "timestamp": "2018-04-07T17:08:00.000Z",
    "topic": "",
    "treatments": Array [],
    "type": "case",
  },
}
`;

exports[`casesReducer returns the initial state 1`] = `Object {}`;

exports[`casesReducer save note 1`] = `
Object {
  "Test Case": Object {
    "_attachments": Object {
      "undefined": Object {
        "content_type": undefined,
        "data": Object {},
      },
    },
    "_id": "",
    "_rev": "",
    "agents": Array [],
    "appKeyId": "",
    "careTeam": Array [],
    "communications": Array [],
    "features": Array [],
    "incidentId": "",
    "investigations": Array [],
    "naturalId": "",
    "notes": Object {
      "Test Note": Object {
        "attachments": Array [
          "undefined",
        ],
        "content": "Test Content",
        "context": "Test Context",
        "id": "Test Note",
        "spi": Object {
          "id": undefined,
          "job": undefined,
          "name": "Test SPI",
          "org": undefined,
          "roles": undefined,
          "siteId": undefined,
        },
        "subcontext": "Test Subcontext",
        "timestamp": "2018-04-07T17:08:00.000Z",
        "title": "Test Title",
      },
    },
    "outcome": "",
    "patient": Object {
      "age": "",
      "ageGroup": "adult",
      "ageUnits": "years",
      "ageYears": "",
      "animalType": "",
      "category": "human",
      "circumstance": "",
      "gender": "",
      "locality": Object {
        "address1": "",
        "address2": "",
        "city": "",
        "country": "",
        "geometry": Object {},
        "isoCountry": "",
        "location": "",
        "name": "",
        "placeId": "",
        "postcode": "",
        "state": "",
        "telephone": "",
      },
      "medicalRecordNumber": null,
      "medicalRecordNumberEnc": "",
      "name": "",
      "nameEnc": "",
      "occupation": "unknown",
      "poisonSeverityScore": "Moderate",
      "pregnant": Object {
        "lactating": "unknown",
        "status": "unknown",
        "trimester": "unknown",
        "trimesterUnknown": null,
      },
      "riskAssessment": "Moderate Risk",
      "weight": "",
      "weightUnits": "kg",
    },
    "referrals": Array [],
    "siteCreated": 1,
    "status": "open",
    "timestamp": "2018-04-07T17:08:00.000Z",
    "topic": "",
    "treatments": Array [],
    "type": "case",
  },
}
`;

exports[`casesReducer update application key id 1`] = `
Object {
  "Test Case": Object {
    "_attachments": Object {},
    "_id": "",
    "_rev": "",
    "agents": Array [],
    "appKeyId": "appKeyId",
    "careTeam": Array [],
    "communications": Array [],
    "features": Array [],
    "incidentId": "",
    "investigations": Array [],
    "naturalId": "",
    "notes": Object {},
    "outcome": "",
    "patient": Object {
      "age": "",
      "ageGroup": "adult",
      "ageUnits": "years",
      "ageYears": "",
      "animalType": "",
      "category": "human",
      "circumstance": "",
      "gender": "",
      "locality": Object {
        "address1": "",
        "address2": "",
        "city": "",
        "country": "",
        "geometry": Object {},
        "isoCountry": "",
        "location": "",
        "name": "",
        "placeId": "",
        "postcode": "",
        "state": "",
        "telephone": "",
      },
      "medicalRecordNumber": null,
      "medicalRecordNumberEnc": "",
      "name": "",
      "nameEnc": "",
      "occupation": "unknown",
      "poisonSeverityScore": "Moderate",
      "pregnant": Object {
        "lactating": "unknown",
        "status": "unknown",
        "trimester": "unknown",
        "trimesterUnknown": null,
      },
      "riskAssessment": "Moderate Risk",
      "weight": "",
      "weightUnits": "kg",
    },
    "referrals": Array [],
    "siteCreated": 1,
    "status": "open",
    "timestamp": "2018-04-07T17:08:00.000Z",
    "topic": "",
    "treatments": Array [],
    "type": "case",
  },
}
`;

exports[`casesReducer update case outcome 1`] = `
Object {
  "Test Case": Object {
    "_attachments": Object {},
    "_id": "",
    "_rev": "",
    "agents": Array [],
    "appKeyId": "",
    "careTeam": Array [],
    "communications": Array [],
    "features": Array [],
    "incidentId": "",
    "investigations": Array [],
    "naturalId": "",
    "notes": Object {},
    "outcome": "New Outcome",
    "patient": Object {
      "age": "",
      "ageGroup": "adult",
      "ageUnits": "years",
      "ageYears": "",
      "animalType": "",
      "category": "human",
      "circumstance": "",
      "gender": "",
      "locality": Object {
        "address1": "",
        "address2": "",
        "city": "",
        "country": "",
        "geometry": Object {},
        "isoCountry": "",
        "location": "",
        "name": "",
        "placeId": "",
        "postcode": "",
        "state": "",
        "telephone": "",
      },
      "medicalRecordNumber": null,
      "medicalRecordNumberEnc": "",
      "name": "",
      "nameEnc": "",
      "occupation": "unknown",
      "poisonSeverityScore": "Moderate",
      "pregnant": Object {
        "lactating": "unknown",
        "status": "unknown",
        "trimester": "unknown",
        "trimesterUnknown": null,
      },
      "riskAssessment": "Moderate Risk",
      "weight": "",
      "weightUnits": "kg",
    },
    "referrals": Array [],
    "siteCreated": 1,
    "status": "open",
    "timestamp": "2018-04-07T17:08:00.000Z",
    "topic": "",
    "treatments": Array [],
    "type": "case",
  },
}
`;

exports[`casesReducer update case status 1`] = `
Object {
  "Test Case": Object {
    "_attachments": Object {},
    "_id": "",
    "_rev": "",
    "agents": Array [],
    "appKeyId": "",
    "careTeam": Array [],
    "communications": Array [
      Object {
        "history": Object {
          "agents": Object {},
          "features": Object {
            "descriptors": "",
            "outcome": "",
            "pastDescriptors": Object {},
            "when": "",
          },
          "investigations": Object {},
          "patient": Object {
            "age": null,
            "ageGroup": null,
            "ageUnits": null,
            "ageYears": null,
            "animalType": null,
            "category": null,
            "circumstance": null,
            "gender": null,
            "localityAddress1": null,
            "localityCity": null,
            "localityCountry": null,
            "localityGeometry": null,
            "localityISOCountry": null,
            "localityLocation": null,
            "localityName": null,
            "localityPlaceId": null,
            "localityPostcode": null,
            "localityState": null,
            "localityTelephone": null,
            "medicalRecordNumber": null,
            "medicalRecordNumberEnc": null,
            "name": null,
            "nameEnc": null,
            "occupation": null,
            "outcome": null,
            "poisonSeverityScore": null,
            "pregnantLactating": null,
            "pregnantStatus": null,
            "pregnantTrimester": null,
            "pregnantTrimesterUnknown": null,
            "riskAssessment": null,
            "topic": null,
            "weight": null,
            "weightUnits": null,
          },
          "treatments": Object {
            "pastTreatments": Object {},
            "referrals": Array [],
            "treatments": Array [],
          },
        },
        "newState": "open",
        "spi": Object {
          "id": undefined,
          "job": undefined,
          "name": "Test SPI",
          "org": undefined,
          "roles": undefined,
          "siteId": undefined,
        },
        "timestamp": "2018-04-07T17:08:00.000Z",
        "type": "status",
      },
    ],
    "features": Array [],
    "incidentId": "",
    "investigations": Array [],
    "naturalId": "",
    "notes": Object {},
    "outcome": "",
    "patient": Object {
      "age": "",
      "ageGroup": "adult",
      "ageUnits": "years",
      "ageYears": "",
      "animalType": "",
      "category": "human",
      "circumstance": "",
      "gender": "",
      "locality": Object {
        "address1": "",
        "address2": "",
        "city": "",
        "country": "",
        "geometry": Object {},
        "isoCountry": "",
        "location": "",
        "name": "",
        "placeId": "",
        "postcode": "",
        "state": "",
        "telephone": "",
      },
      "medicalRecordNumber": null,
      "medicalRecordNumberEnc": "",
      "name": "",
      "nameEnc": "",
      "occupation": "unknown",
      "poisonSeverityScore": "Moderate",
      "pregnant": Object {
        "lactating": "unknown",
        "status": "unknown",
        "trimester": "unknown",
        "trimesterUnknown": null,
      },
      "riskAssessment": "Moderate Risk",
      "weight": "",
      "weightUnits": "kg",
    },
    "referrals": Array [],
    "siteCreated": 1,
    "status": "open",
    "timestamp": "2018-04-07T17:08:00.000Z",
    "topic": "",
    "treatments": Array [],
    "type": "case",
  },
}
`;

exports[`casesReducer update case topic 1`] = `
Object {
  "Test Case": Object {
    "_attachments": Object {},
    "_id": "",
    "_rev": "",
    "agents": Array [],
    "appKeyId": "",
    "careTeam": Array [],
    "communications": Array [],
    "features": Array [],
    "incidentId": "",
    "investigations": Array [],
    "naturalId": "",
    "notes": Object {},
    "outcome": "",
    "patient": Object {
      "age": "",
      "ageGroup": "adult",
      "ageUnits": "years",
      "ageYears": "",
      "animalType": "",
      "category": "human",
      "circumstance": "",
      "gender": "",
      "locality": Object {
        "address1": "",
        "address2": "",
        "city": "",
        "country": "",
        "geometry": Object {},
        "isoCountry": "",
        "location": "",
        "name": "",
        "placeId": "",
        "postcode": "",
        "state": "",
        "telephone": "",
      },
      "medicalRecordNumber": null,
      "medicalRecordNumberEnc": "",
      "name": "",
      "nameEnc": "",
      "occupation": "unknown",
      "poisonSeverityScore": "Moderate",
      "pregnant": Object {
        "lactating": "unknown",
        "status": "unknown",
        "trimester": "unknown",
        "trimesterUnknown": null,
      },
      "riskAssessment": "Moderate Risk",
      "weight": "",
      "weightUnits": "kg",
    },
    "referrals": Array [],
    "siteCreated": 1,
    "status": "open",
    "timestamp": "2018-04-07T17:08:00.000Z",
    "topic": "New Topic",
    "treatments": Array [],
    "type": "case",
  },
}
`;

exports[`casesReducer update case type 1`] = `
Object {
  "Test Case": Object {
    "_attachments": Object {},
    "_id": "",
    "_rev": "",
    "agents": Array [],
    "appKeyId": "",
    "careTeam": Array [],
    "communications": Array [],
    "features": Array [],
    "incidentId": "",
    "investigations": Array [],
    "naturalId": "",
    "notes": Object {},
    "outcome": "",
    "patient": Object {
      "age": "",
      "ageGroup": "adult",
      "ageUnits": "years",
      "ageYears": "",
      "animalType": "",
      "category": "human",
      "circumstance": "",
      "gender": "",
      "locality": Object {
        "address1": "",
        "address2": "",
        "city": "",
        "country": "",
        "geometry": Object {},
        "isoCountry": "",
        "location": "",
        "name": "",
        "placeId": "",
        "postcode": "",
        "state": "",
        "telephone": "",
      },
      "medicalRecordNumber": null,
      "medicalRecordNumberEnc": "",
      "name": "",
      "nameEnc": "",
      "occupation": "unknown",
      "poisonSeverityScore": "Moderate",
      "pregnant": Object {
        "lactating": "unknown",
        "status": "unknown",
        "trimester": "unknown",
        "trimesterUnknown": null,
      },
      "riskAssessment": "Moderate Risk",
      "weight": "",
      "weightUnits": "kg",
    },
    "referrals": Array [],
    "siteCreated": 1,
    "status": "open",
    "timestamp": "2018-04-07T17:08:00.000Z",
    "topic": "",
    "treatments": Array [],
    "type": "hoax",
  },
}
`;
